contracts/StablecoinConverter.sol

```diff
@@ -167,4 +174,49 @@ contract StablecoinConverter is EpochTokenLocker {
         }
         revert("Price not provided for token");
     }
+
+    function documentTrades(
+        uint batchIndex,
+        address[] memory owner,  //tradeData is submitted as arrays
+        uint16[] memory orderId,
+        uint128[] memory volume,
+        uint128[] memory prices,
+        uint16[] memory tokenIdForPrice
+    ) internal {
+        previousSolutionBatchId = batchIndex;
+        uint len = owner.length;
+        for (uint i = 0; i < len; i++) {
+            previousSolutionTrades.push(TradeData({
+                owner: owner[i],
+                orderId: orderId[i],
+                volume: volume[i]
+            }));
+        }
+        len = prices.length;
+        for (uint i = 0; i < len; i++) {
+            previousSolutionPrices.push(prices[i]);
+            previousSolutionTokenIdForPrice.push(tokenIdForPrice[i]);
+        }
+    }
+
+    function undoPreviousSolution(uint batchIndex) internal {
+        if (previousSolutionBatchId == batchIndex) {
+            for (uint i = 0; i < previousSolutionTrades.length; i++) {
+                Order memory order = orders[previousSolutionTrades[i].owner][previousSolutionTrades[i].orderId];
```

Comment:
Can we store `previousSolutionTrades[i]` in a local variable for readability? This code is very hard to read.